extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/TaskAdapter.java

```diff
@@ -19,15 +19,17 @@
 
 package org.apache.druid.k8s.overlord.common;
 
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.batch.v1.Job;
 import org.apache.druid.indexing.common.task.Task;
 
 import java.io.IOException;
 
-public interface TaskAdapter<K, V>
+public interface TaskAdapter
 {
 
-  V fromTask(Task task, PeonCommandContext context) throws IOException;
```

Comment:
i think there was some idea that we would want to leave open the option of non K8s Task Runners (e.g. fargate) which would imply the need for non K8s TaskAdapters, is there a need to override this class? wouldn't K8sTaskAdapter implements TaskAdapter<Job, Pod> still work here?

Reply:
Realizing TaskAdapter is actually a K8s only concept so this comment is not relevant.

I do think TaskAdapter being only for k8s to be a little confusing though

Reply:
We would implement Fargate within a FargateTaskRunner.  The `TaskAdapter` concept is a KubernetesTaskRunner only concept not a Druid one.

Reply:
why would Fargate be implemented through a different runner? doesn't fargate offer k8s api?

Reply:
I'm unfamiliar with the api that Fargate provides.  If it does offer a k8s api then yes we could continue to use the KubernetesTaskRunner.